### PR TITLE
feat: Add Alert Component

### PR DESCRIPTION
**Pull request by CharlieSu**

### CODE REVIEW

**Comment by CharlieSu** [comment]
@osbornm  - Can you help me figure out this build error?  I didn't touch this component and locally my system sees this import but fails to build.

https://github.com/Hyphen/hyphen-components/actions/runs/7718159289/job/21038831259?pr=4#step:5:44

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,6 @@
+export * from './components/Alert/Alert';
 export * from './components/Box/Box';
+export * from './components/Button/Button';
+export * from './components/Heading/Heading';
+export * from './components/Icon/Icon';
+export * from './components/Spinner/Spinner';
```
Comment (osbornm) [inline src/index.ts:6]: Crap good call here! totally been forgetting to do this.

**Comment by osbornm** [comment]
We are also using semantic release here just like with design tokens (sorry not documented) so you'll need to name this PR correctly. see: https://github.com/Hyphen/hyphen-design-tokens?tab=readme-ov-file#releases

**Comment by osbornm** [comment]
The issue with the button component is a difference in storybook build and dts build I will fix that in a separate branch/pr. After commenting that out you had another build error I fixed.

**Review by osbornm — APPROVED** [review]

**Pull request merged**